src/sequences/Formula.ts

```diff
@@ -101,22 +101,65 @@ Plus the standard parameters for all formulas:
 !}
 **/
 
+const maxWarns = 3
 class Formula extends Cached(paramDesc) {
     static category = 'Formula'
     name = `${Formula.category}: ${paramDesc.formula.default}`
     static description = 'A sequence defined by a formula in n'
+    nErrors = 0
 
     initialize(): void {
         super.initialize()
         this.name = 'Formula: ' + this.formula.source
+        this.nErrors = 0
     }
 
     calculate(n: bigint) {
-        const result = this.formula.compute(Number(n))
+        let result = 0
+        let resultType = ''
+        try {
+            const rawResult = this.formula.compute(Number(n))
+            resultType = typeof rawResult
+            result = math.floor(rawResult)
+            if (isNaN(result)) {
+                result = 0
+                throw new Error(
+                    `result '${rawResult}' of calculation is `
+                        + `not a number.`
+                )
+            }
+        } catch (err: unknown) {
+            this.nErrors++
+            if (this.nErrors < maxWarns) {
+                let message =
+                    err instanceof Error ? err.message : 'of unkown error.'
+                if (resultType && message.includes('convert')) {
+                    message = message.replace(
+                        'convert',
+                        `convert ${resultType} value`
+                    )
+                }
+                this.statusOf.formula.addWarning(
+                    `value for n=${n} set to ${result} because ${message}`
+                )
+            } else if (this.nErrors === maxWarns) {
+                this.statusOf.formula.addWarning(
+                    `[1 additional warning discarded]`
+                )
+            } else {
+                // replace discarded message
+                const ensure = this.statusOf.formula.warnings.pop()
+                console.assert(ensure && ensure.endsWith('discarded]'))
+                this.statusOf.formula.addWarning(
+                    `[${this.nErrors - maxWarns + 1} additional `
+                        + 'warnings discarded]'
+                )
```

Comment:
I'm not sure why, but when I tested this by beginning with `typeOf(n)` it gave the expected error, then I expanded this to `typeOf(matrix([n,n]))` and the following happened:  the warnings were all hidden (but apparently present...).  Bug?
![Screenshot from 2024-11-16 21-19-14](https://github.com/user-attachments/assets/6a2fee71-2048-4f6a-bc51-932afb428b00)

Reply:
Hmm yes seems like nErrors is not getting reset even though the warnings list was apparently cleared. Will investigate.

Reply:
Hmmm can't reproduce. I used formula `typeOf(n)` and got the expected warnings. I tried inserting my cursor after the opening paren and typing `matrix([`, then skipping over the `n`, then typing the `])`, but that left the warnings looking totally fine. I also tried having `matrix([n,n])` in my clipboard and pasting it over the `n`, but again all looked well. So I think I need the exact URL you started with, and the exact sequence of keystrokes/pastes/whatever to get it into the shown state to debug.

Reply:
Go to http://localhost:5173/?name=Formula&viz=ModFill&modDimension=1000&seq=Formula&formula=typeOf%28matrix%28%5Bn%2Cn%5D%29%29 and then backspace the [ and replace it.

Reply:
Indeed, I can now reproduce it. Will investigate.

Reply:
OK, that was a subtle one. How about now?

Reply:
Now the URL I sent you shows too many warnings.
![Screenshot from 2024-11-17 00-03-02](https://github.com/user-attachments/assets/a5ddce00-5e14-4338-b4d6-c38b496c3d69)

Reply:
Hunh. It only happens on page load/reload. If I wipe out some totally unrelated formula and then type in this one, it's totally fine. Ugh. Will have to dive deeper tomorrow.

Reply:
Something to do with Vue?

Reply:
OK, gave up on trying to sync a property of the Formula class and the warnings on the list. Now I just use the warnings themselves to keep track and all seems well. What do you think?

Reply:
Looks like it's working.  Question -- what controls how many n's it tries (and therefore how many warnings are suppressed).  For example when it says "n=1280 set to 0 because Cannot convert string value "DenseMatrix" to a number" why 1280?

Reply:
In fact, weird behaviour, this n=whatever increments when I do the erase + replace left square bracket (as I described for the original bug).  Probably doesn't matter...

Reply:
well, I'm impressed, it let me type this: det(matrix([[n,n],[n^2,n-1]]))*n^n^n^n^n^n^n^n^n^n^n^n^n^n^n^n*2*3*4*5*6+combinations(n,2)

Reply:
sorry, git ate some of the symbols

Reply:
> weird behaviour, this n=whatever increments when I do the erase + replace left square bracket (as I described for the original bug). Probably doesn't matter...

Well, actually it is not so weird. You're mucked around with the formula, but gone back to what the formula was before. So frontscope is clever enough to realize that this is the _same_ Sequence that it was before, but that mucking around it interprets as a request for more values, so it caches some more values of that same Sequence. That's why the _n_ starts much larger this time, it's adding more values to the end of what's already cached. Make sense?

Reply:
> it let me type this

Did it actually compute any values successfully? I guess n=1 is ok but it seems like that expression should overflow at n=2...

Reply:
Let me know if there's anything else I need to do on this.

Reply:
> Did it actually compute any values successfully? I guess n=1 is ok but it seems like that expression should overflow at n=2...

Well that's hard to know, it did output something that modfill was willing to plot.  Maybe we should file an issue to investigate whether overflows are silently giving garbage.